force_torque_sensor_broadcaster/src/force_torque_sensor_broadcaster.cpp

```diff
@@ -53,18 +47,19 @@ controller_interface::CallbackReturn ForceTorqueSensorBroadcaster::on_init()
 controller_interface::CallbackReturn ForceTorqueSensorBroadcaster::on_configure(
   const rclcpp_lifecycle::State & /*previous_state*/)
 {
-  sensor_name_ = get_node()->get_parameter("sensor_name").as_string();
-  interface_names_[0] = get_node()->get_parameter("interface_names.force.x").as_string();
-  interface_names_[1] = get_node()->get_parameter("interface_names.force.y").as_string();
-  interface_names_[2] = get_node()->get_parameter("interface_names.force.z").as_string();
-  interface_names_[3] = get_node()->get_parameter("interface_names.torque.x").as_string();
-  interface_names_[4] = get_node()->get_parameter("interface_names.torque.y").as_string();
-  interface_names_[5] = get_node()->get_parameter("interface_names.torque.z").as_string();
+  if (!param_listener_)
+  {
+    RCLCPP_ERROR(get_node()->get_logger(), "Parameters listener failed to construct in on_init");
+    return controller_interface::CallbackReturn::ERROR;
+  }
```

Comment:
Is this really needed here? Shouldn't line 36 already fail?

Reply:
It guards against on_configure being called and resulting in a segfault in the case where on_init failed and param_listener_ is not yet valid.

Reply:
```suggestion
```